client/asset/dcr/dcr.go

```diff
@@ -1348,22 +1352,25 @@ func (dcr *ExchangeWallet) tryFund(utxos []*compositeUTXO, enough func(sum uint6
 			if len(okUTXOs) == 0 {
 				return false, nil
 			}
-			// On each loop, find the smallest UTXO that is enough.
-			for _, txout := range okUTXOs {
-				if enough(sum, size, txout) {
-					if err = addUTXO(txout); err != nil {
-						return false, err
-					}
-					return true, nil
-				}
+
+			// Check if the largest output is too small.
+			lastUTXO := okUTXOs[len(okUTXOs)-1]
+			if !isEnoughWith(lastUTXO) {
+				addUTXO(lastUTXO)
```

Comment:
We check the error from `addUTXO` below, and both places previously.

Although thinking about this, one utxo with a screwed up txid or redeem script would ideally not break utxo selection entirely.  How about at the top of `tryFund` we do a last minute check and re-parse of the `compositeUTXO` slice before the `tryUTXOs` calls, generating a `[]parsedUTXO` with a local type like

```go
type parsedUTXO struct {
	hash         *chainhash.Hash
	vout         uint32
	redeemScript []byte
	amt          uint64
	tree         int8
	inSize       uint32
}
```

And refactor `addUTXO` to take that type and return no error.

Reply:
We could even go all the way back to `spendableUTXOs`, but I would kinda like to avoid doing that.

Reply:
Just checked the error for now.